pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -405,6 +408,48 @@ func (configgen *ConfigGeneratorImpl) buildGatewayInboundHTTPRouteConfig(
 	return out
 }
 
+// mergeIngress will append rules from Ingress-generated virtual services to existing virtual services.
+// The ingress can be created dynamically, and only has very limitted capabilities -
+// [host]/path->backend configuration
+func mergeIngress(configs []model.Config) []model.Config {
+	configsOut := []model.Config{}
+	hostsToVS := map[string]*networking.VirtualService{}
+
+	// Ignore ingress resources, build a map of hosts to (real) VirtualServices.
+	for _, c := range configs {
+		vs := c.Spec.(*networking.VirtualService)
+		for _, h := range vs.Hosts {
+			if !strings.HasSuffix(c.Name, model.IstioIngressGatewayName) {
+				vse, e := hostsToVS[h]
+				if e {
+					log.Warnf("Duplicated hostname %s OLD:%v NEW:%v", h, vse, vs)
+					continue
+				}
+				hostsToVS[h] = vs
+				configsOut = append(configsOut, c)
+			}
+		}
+	}
+
+	for _, c := range configs {
+		vs := c.Spec.(*networking.VirtualService)
+		for _, h := range vs.Hosts {
+			if strings.HasSuffix(c.Name, model.IstioIngressGatewayName) {
+				vse, e := hostsToVS[h]
+				if e {
+					vse.Http = append(vse.Http, vs.Http...)
+					log.Infof("Merging ingress to vs %s %s", h, c.Name)
```

Comment:
The append was already done in the conversion. Why iterate over this again?

Reply:
Good point, will remove ( I implemented this first - then figured it's easy to merge during conversion ).